transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SchemaManager.java

```diff
@@ -155,12 +147,16 @@ public void onOpen(SQLiteDatabase db) {
   }
 
   private void upgrade(SQLiteDatabase db, int fromVersion, int toVersion) {
-    for (Migration m : INCREMENTAL_MIGRATIONS) {
-      m.upgrade(db, fromVersion, toVersion);
+    int highestVersion = INCREMENTAL_MIGRATIONS.size();
+
+    for (int version = 1; version <= highestVersion; version++) {
```

Comment:
We might want to be more strict about this logic, e.g. throw an exception if `migrations.size() < toVersion` meaning we are missing some migrations for whatever reason. Additionally we should be able to iterate in the following range instead of from `1 - size()`: `int i = fromVersion; i <= toVersion`, this should also remove the need to do the if check below, wdyt?

Reply:
Just confirming. I think it should be < and not <=
```
    for (int version = fromVersion; version < toVersion; version++) {
```